zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/GroupControllerTest.java

```diff
@@ -0,0 +1,256 @@
+
+    final var expectedBody = ProblemDetail.forStatusAndDetail(HttpStatus.BAD_REQUEST, message);
+    expectedBody.setTitle(IllegalArgumentException.class.getName());
+    expectedBody.setInstance(URI.create("/v2/groups"));
+
+    webClient
+        .post()
+        .uri("/v2/groups")
+        .accept(MediaType.APPLICATION_JSON)
+        .contentType(MediaType.APPLICATION_JSON)
+        .bodyValue(camundaGroupRequest)
+        .exchange()
+        .expectStatus()
+        .isBadRequest()
+        .expectBody(ProblemDetail.class)
+        .isEqualTo(expectedBody);
+  }
+
+  @Test
+  void updateGroupShouldReturnUpdatedGroup() {
+    final CamundaGroup camundaGroup = new CamundaGroup(1L, "demoChanged");
+
+    final CamundaGroupRequest camundaGroupRequest = new CamundaGroupRequest();
+    camundaGroupRequest.setId(1L);
+    camundaGroupRequest.setName("demoChanged");
+
+    final CamundaGroupResponse camundaGroupResponse = new CamundaGroupResponse();
+    camundaGroupResponse.setId(1L);
+    camundaGroupResponse.setName("demoChanged");
+
+    when(groupService.updateGroup(1L, camundaGroup)).thenReturn(camundaGroup);
+
+    webClient
+        .put()
+        .uri("/v2/groups/1")
+        .accept(MediaType.APPLICATION_JSON)
+        .contentType(MediaType.APPLICATION_JSON)
+        .bodyValue(camundaGroupRequest)
+        .exchange()
+        .expectStatus()
+        .is2xxSuccessful()
+        .expectBody(CamundaGroupResponse.class)
+        .isEqualTo(camundaGroupResponse);
+
+    verify(groupService, times(1)).updateGroup(1L, camundaGroup);
+  }
+
+  @Test
+  void updateGroupByIdThrowsExceptionWhenServiceThrowsException() {
+    final String message = "message";
+
+    final CamundaGroup camundaGroup = new CamundaGroup(1L, "demoChanged");
+
+    final CamundaGroupRequest camundaGroupRequest = new CamundaGroupRequest();
+    camundaGroupRequest.setId(1L);
+    camundaGroupRequest.setName("demoChanged");
+
+    when(groupService.updateGroup(1L, camundaGroup))
+        .thenThrow(new IllegalArgumentException(message));
+
+    final var expectedBody = ProblemDetail.forStatusAndDetail(HttpStatus.BAD_REQUEST, message);
+    expectedBody.setTitle(IllegalArgumentException.class.getName());
+    expectedBody.setInstance(URI.create("/v2/groups/1"));
+
+    webClient
+        .put()
+        .uri("/v2/groups/1")
+        .accept(MediaType.APPLICATION_JSON)
+        .contentType(MediaType.APPLICATION_JSON)
+        .bodyValue(camundaGroupRequest)
+        .exchange()
+        .expectStatus()
+        .isBadRequest()
+        .expectBody(ProblemDetail.class)
+        .isEqualTo(expectedBody);
+  }
+
+  @Test
+  void deleteGroupShouldDeleteGroup() {
+
+    webClient
+        .delete()
+        .uri("/v2/groups/1")
+        .accept(MediaType.APPLICATION_JSON)
+        .exchange()
+        .expectStatus()
+        .is2xxSuccessful();
+
+    verify(groupService, times(1)).deleteGroupById(1L);
+  }
+
+  @Test
+  void searchGroupsShouldReturnGroups() {
```

Comment:
🔧 As mentioned in the other PR, it would be nice to have a test case that verifies what happens if an exception is thrown in the service when we invoke the search.

Reply:
In an effort to make progress I will merge this PR as is but will look to include this in a later PR